### PR TITLE
docs: restore rust plugin content on `main`

### PR DESCRIPTION
**Pull request by StephenBarlow**
This effectively lands https://github.com/apollographql/router/pull/2838 directly onto our `main` branch which is the deployed version of our documentation.  We can close #2838 after that as `main` will merge back to `dev` per the usual post-release process.

### CODE REVIEW

**Comment by abernix** [comment]
Closing and re-opening to trigger CI, maybe.

File: docs/source/customizations/custom-binary.mdx [hunk 1]
```diff
@@ -4,9 +4,14 @@ title: Creating a custom Apollo Router binary
 
 import ElasticNotice from '../../shared/elastic-notice.mdx';
 
-> ⚠️ **Native Rust plugins are not recommended!** Instead, you should extend your router with a [Rhai script](./rhai/) or [external coprocessor](./coprocessor/).
-
-To use [custom plugins](./native/) that aren't bundled with the default Apollo Router distribution, you need to compile your own custom router binary. This page walks you through creating that binary from scratch and adding a plugin to it.
+> ⚠️ **Native Rust plugins are an advanced feature of the Apollo Router.** We recommend _against_ creating them for the following reasons:
```
Comment (abernix) [inline docs/source/customizations/custom-binary.mdx:7]:
```suggestion
> ⚠️ **Native Rust plugins are an advanced feature of the Apollo Router.** We _do not recommend_ creating them for the following reasons:
```

File: docs/source/customizations/native.mdx [hunk 2]
```diff
@@ -3,7 +3,14 @@ title: Writing native Rust plugins for the Apollo Router
 description: Extend the Apollo Router with custom Rust code
 ---
 
-> ⚠️ **Native Rust plugins are not recommended!** Instead, you should extend your router with a [Rhai script](./rhai/) or [external coprocessor](./coprocessor/).
+> ⚠️ **Native Rust plugins are an advanced feature of the Apollo Router.** We recommend _against_ creating them for the following reasons:
```
Comment (abernix) [inline docs/source/customizations/native.mdx:6]:
```suggestion
> ⚠️ **Native Rust plugins are an advanced feature of the Apollo Router.** We _do not recommend_ creating them for the following reasons:
```

**Review by abernix — APPROVED** [review]

**Review by prasek — APPROVED** [review]
lgtm

**Pull request merged**